etcd/etcd.go

```diff
@@ -214,7 +210,14 @@ func (c *client) RegisterConfigCallback(ctx context.Context, key string, uniqueI
 func (c *client) DeregisterConfig(key string, uniqueID int64) {
 	m.Lock()
 	clientKey := key + "/" + strconv.FormatInt(uniqueID, 10)
-	cancel := ctxMap[clientKey]
+	cancel := c.cancelMap[clientKey]
 	cancel()
 	m.Unlock()
 }
+
+func (c *client) register(key string, uniqueID int64, cancel context.CancelFunc) {
```

Comment:
Since it's integrated into the client, so a more specific name like `registerCancelFunc` will be better.